### PR TITLE
Provide compat monitor route

### PR DESCRIPTION
**Pull request by eduardoj**
The mandatory 'project' parameter for the project monitor page was introduced in f795712.

Routes like `/project/monitor?project=my_project` will now be redirected to `/project/monitor/my_project`, so users can keep the same monitor routes they already had, and will not get an 404 error.

Thanks to @darix for reporting it.

### CODE REVIEW

File: src/api/config/routes.rb [hunk 1]
```diff
@@ -287,6 +287,10 @@ def self.public_or_about_path?(request)
       post 'project/remove_person/:project' => :remove_person, constraints: cons
       post 'project/remove_group/:project' => :remove_group, constraints: cons
       get 'project/monitor/:project' => :monitor, constraints: cons, as: 'project_monitor'
+      # compat route
+      get 'project/monitor', to: redirect { |_path_parameters, request|
+        "/project/monitor/#{request.query_parameters[:project]}?#{request.query_parameters.except(:project).to_param}"
```
Comment (Ana06) [inline src/api/config/routes.rb]: where is this route actually used? What about stop using it instead of redirecting? :eyes:

Comment (darix) [reply src/api/config/routes.rb]: What about not breaking people's bookmarks for no reason?

**Comment by dmarcoux** [comment]
For some reasons, CircleCI didn't report back about `checkout_code` and `backend_test`. They both passed.

File: src/api/config/routes.rb [hunk 2]
```diff
@@ -287,6 +287,10 @@ def self.public_or_about_path?(request)
       post 'project/remove_person/:project' => :remove_person, constraints: cons
       post 'project/remove_group/:project' => :remove_group, constraints: cons
       get 'project/monitor/:project' => :monitor, constraints: cons, as: 'project_monitor'
+      # compat route
```
Comment (dmarcoux) [inline src/api/config/routes.rb]: I find the comment `compat route` not so great. I would use something like `Backward compatible route`, but I don't want to block the PR for this.

File: src/api/config/routes.rb (hunk 1 above)
Comment (Ana06) [reply src/api/config/routes.rb]: @darix 

> What about not breaking people's bookmarks for no reason?

This comment was for @eduardoj, not for you. Fixing the Bookmarks doesn't means we have to do it in this way. Sorry that this broke for you, but this is IMO not the correct solution. We can come up with a fast solution if this is a priority bug (BTW, where is the bug report?), but this shouldn't stay like that.

Comment (Ana06) [reply src/api/config/routes.rb]: I got wrong was this what about

File: src/api/config/routes.rb (hunk 2 above)
Comment (Ana06) [reply src/api/config/routes.rb]: I would add a `TODO` to remove this route in the next big OBS release

**Comment by eduardoj** [comment]
@Ana06:

> I would add a `TODO` to remove this route in the next big OBS release

`TODO` added. Note that in several other parts of the code it is used `FIXME`, `FIXME3.0`, or `FIXME: 3.0`.

File: src/api/config/routes.rb (hunk 2 above)
Comment (eduardoj) [reply src/api/config/routes.rb]: @dmarcoux:

> I find the comment `compat route` not so great. I would use something like `Backward compatible route`, but I don't want to block the PR for this.

I added a commit to change this comment, and all others in the file, to `For backward compatibility`.

File: src/api/config/routes.rb [hunk 3]
```diff
@@ -287,6 +287,10 @@ def self.public_or_about_path?(request)
       post 'project/remove_person/:project' => :remove_person, constraints: cons
       post 'project/remove_group/:project' => :remove_group, constraints: cons
       get 'project/monitor/:project' => :monitor, constraints: cons, as: 'project_monitor'
+      # For backward compatibility
+      get 'project/monitor', to: redirect { |_path_parameters, request|
+        "/project/monitor/#{request.query_parameters[:project]}?#{request.query_parameters.except(:project).to_param}"
```
Comment (dmarcoux) [inline src/api/config/routes.rb]: This leaves an extra `?` when there are no query parameters beside `project`. So if you go to `project/monitor?project=home:Admin`, you are redirected to `/project/monitor/home:Admin?`

Comment (eduardoj) [reply src/api/config/routes.rb]: Thanks. Fixed. Please have another look.

File: src/api/config/routes.rb [hunk 4]
```diff
@@ -287,6 +287,12 @@ def self.public_or_about_path?(request)
       post 'project/remove_person/:project' => :remove_person, constraints: cons
       post 'project/remove_group/:project' => :remove_group, constraints: cons
       get 'project/monitor/:project' => :monitor, constraints: cons, as: 'project_monitor'
+      # For backward compatibility
+      get 'project/monitor', to: redirect { |_path_parameters, request|
+        url_string = request.query_parameters.except(:project).to_param
+        url_string = '?' << url_string unless url_string.empty?
+        "/project/monitor/#{request.query_parameters[:project]}?#{url_string}"
```
Comment (dmarcoux) [inline src/api/config/routes.rb]: There is still an extra`?` on this line. Please, remove it.

Comment (eduardoj) [reply src/api/config/routes.rb]: My fault. Thanks. Done.

**Pull request merged**